tests/testthat/test-eval-relocate.R

```diff
@@ -0,0 +1,192 @@
+  x <- c(a = 1, x = 1, b = 1, z = 1, y = 1)
+
+  expect_named(
+    relocate_loc(x, c(x, y, z), before = x),
+    c("a", "x", "y", "z", "b")
+  )
+  expect_named(
+    relocate_loc(x, c(x, y, z), after = last_col()),
+    c("a", "b", "x", "y", "z")
+  )
+  expect_named(
+    relocate_loc(x, c(x, a, z)),
+    c("x", "a", "z", "b", "y")
+  )
+})
+
+test_that("allows for renaming (tidyverse/dplyr#5569)", {
+  x <- c(a = 1, b = 1, c = 1)
+
+  expect_named(
+    relocate_loc(x, c(new = b)),
+    c("new", "a", "c")
+  )
+  expect_named(
+    relocate_loc(x, c(new = b), after = c),
+    c("a", "c", "new")
+  )
+})
+
+test_that("can't supply both `before` and `after`", {
+  expect_snapshot(error = TRUE, {
+    relocate_loc(c(x = 1), before = 1, after = 1)
+  })
+  expect_snapshot(error = TRUE, {
+    relocate_loc(c(x = 1), before = 1, after = 1, before_arg = ".before", after_arg = ".after")
+  })
+})
+
+test_that("can't relocate with out-of-bounds variables by default", {
+  x <- c(a = 1, b = 2)
+
+  expect_snapshot({
+    (expect_error(relocate_loc(x, c)))
+    (expect_error(relocate_loc(x, a, before = c)))
+    (expect_error(relocate_loc(x, a, after = c)))
+  })
+})
+
+test_that("can relocate with out-of-bounds variables in `expr` if `strict = FALSE`", {
+  x <- c(a = 1, b = 2)
+
+  expect_identical(relocate_loc(x, c, strict = FALSE), c(a = 1L, b = 2L))
+  expect_identical(relocate_loc(x, c(d = b, e = c), strict = FALSE), c(d = 2L, a = 1L))
+
+  # But still not with OOB variables in `before` or `after`
+  expect_snapshot({
+    (expect_error(relocate_loc(x, a, before = c, strict = FALSE)))
+    (expect_error(relocate_loc(x, a, after = c, strict = FALSE)))
+  })
+})
+
+test_that("accepts name spec", {
+  x <- c(a = 1, b = 2, c = 3)
+
+  expect_identical(
+    relocate_loc(x, c(foo = c(c, a)), name_spec = "{outer}_{inner}"),
+    c(foo_1 = 3L, foo_2 = 1L, b = 2L)
+  )
+  expect_identical(
+    relocate_loc(x, c(foo = c(bar = c, baz = a)), name_spec = "{outer}_{inner}"),
+    c(foo_bar = 3L, foo_baz = 1L, b = 2L)
+  )
+})
+
+test_that("can forbid rename syntax", {
+  x <- c(a = 1, b = 2, c = 3)
+
+  expect_snapshot({
+    (expect_error(relocate_loc(x, c(foo = b), allow_rename = FALSE)))
+    (expect_error(relocate_loc(x, c(b, foo = b), allow_rename = FALSE)))
+  })
+
+  expect_named(relocate_loc(x, c(c, b), allow_rename = FALSE), c("c", "b", "a"))
+})
+
+test_that("can forbid empty selections", {
+  x <- c(a = 1, b = 2, c = 3)
+
+  expect_snapshot({
+    (expect_error(relocate_loc(x, allow_empty = FALSE)))
+    (expect_error(relocate_loc(mtcars, integer(), allow_empty = FALSE)))
+    (expect_error(relocate_loc(mtcars, starts_with("z"), allow_empty = FALSE)))
+  })
+})
+
+test_that("`before` and `after` forbid renaming", {
+  x <- c(a = 1, b = 2, c = 3)
+
+  expect_snapshot({
+    (expect_error(relocate_loc(x, b, before = c(new = c))))
+    (expect_error(relocate_loc(x, b, after = c(new = c))))
+  })
+})
```

Comment:
These ignore `allow_rename` and always set it to `FALSE`, because it would have no effect anyways